src/firecracker/examples/uffd/uffd_utils.rs

```diff
@@ -208,6 +208,43 @@ impl Runtime {
         }
     }
 
+    fn peer_process_credentials(&self) -> libc::ucred {
+        let mut creds: libc::ucred = libc::ucred {
+            pid: 0,
+            gid: 0,
+            uid: 0,
+        };
+        let mut creds_size = size_of::<libc::ucred>() as u32;
+        let ret = unsafe {
+            libc::getsockopt(
+                self.stream.as_raw_fd(),
+                libc::SOL_SOCKET,
+                libc::SO_PEERCRED,
+                &mut creds as *mut _ as *mut _,
+                &mut creds_size as *mut libc::socklen_t,
+            )
+        };
+        if ret != 0 {
+            panic!("Failed to get peer process credentials");
+        }
+        creds
+    }
+
+    pub fn install_panic_hook(&self) {
+        let peer_creds = self.peer_process_credentials();
+
+        let default_panic_hook = std::panic::take_hook();
+        std::panic::set_hook(Box::new(move |panic_info| {
+            let r = unsafe { libc::kill(peer_creds.pid, libc::SIGKILL) };
```

Comment:
What is the utility of the change? Is there a way to make use of that in integration tests?

Reply:
This is mainly to make integration tests that are due do the UFFD handler crashing easier to debug. Right now, they just show up as pytest timeouts at 300s (e.g. the intermittent failures in https://github.com/firecracker-microvm/firecracker/pull/4601, which had us stumped for 6 months). With this, Firecracker gets killed, and we get a message later during teardown at least pointing us towards the possibility of the UFFD handler having died.